spec/controllers/admin/books_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require "rails_helper"
+include SessionsHelper
+RSpec.describe Admin::BooksController, type: :controller do
+  let(:admin1){FactoryBot.create :user, role: "admin"}
+  let!(:book_3){FactoryBot.create :book}
+  let!(:book_4){FactoryBot.create :book}
+
+  before do
+    log_in admin1
+  end
+
+  describe "GET #index" do
+    context "when return all book" do
```

Comment:
Nếu nó ko có ngữ cảnh cụ thể thì bỏ context đi em 
Dùng cho linh hoạt nhé, ko cần đầy đủ describer, context, it nhưng cần phải hợp lí nhé